google/cloud/bigquery_storage_v1/reader.py

```diff
@@ -503,22 +528,34 @@ def from_read_session(read_session):
                 "Unsupported schema type in read_session: {0}".format(schema_type)
             )
 
+    @staticmethod
+    def from_read_rows_response(message):
+        schema_type = message._pb.WhichOneof("schema")
+        if schema_type == "avro_schema":
+            return _AvroStreamParser(message)
+        elif schema_type == "arrow_schema":
+            return _ArrowStreamParser(message)
+        else:
+            raise TypeError(
+                "Unsupported schema type in message: {0}".format(schema_type)
+            )
+
 
 class _AvroStreamParser(_StreamParser):
     """Helper to parse Avro messages into useful representations."""
 
-    def __init__(self, read_session):
+    def __init__(self, message):
         """Construct an _AvroStreamParser.
 
         Args:
-            read_session (google.cloud.bigquery_storage_v1.types.ReadSession):
-                A read session. This is required because it contains the schema
-                used in the stream messages.
+            message (google.cloud.bigquery_storage_v1.types.ReadRowsResponse):
+                The first message of data from a read rows stream, which
+                contains the schema.
```

Comment:
Hmm, `_StreamParser.from_read_session` passes in a read session instance -- is that going to be an issue, here?  Or are read sessions and messages always substitutable, at least for purposes of schema discovery?

Reply:
They are substitutable. Both have a `schema` "oneof" with the same fields. Though I should probably document that.